orchestrator/careplancontributor/applaunch/zorgplatform/handle_create_hcp_rst.go

```diff
@@ -143,6 +172,38 @@ func (s *Service) createSAMLAssertion(launchContext *LaunchContext) (*etree.Elem
 	return assertion, nil
 }
 
+func (s *Service) createSAMLAssertionForApplication(launchContext *LaunchContext) (*etree.Element, error) {
+
+	assertion, err := s.createSAMLAssertion(launchContext)
+	if err != nil {
+		return nil, err
+	}
+
+	// Replace the PurposeOfUse attribute in the assertion
+	attrValue := assertion.FindElement(".//Attribute[@Name='urn:oasis:names:tc:xspa:1.0:subject:purposeofuse']//AttributeValue/PurposeOfUse")
+	if attrValue == nil {
+		return nil, fmt.Errorf("PurposeOfUse element not found in assertion")
+	}
+	attrValue.CreateAttr("code", "OPERATIONS")
+
+	// Replace the Role attribute in the assertion
+	roleAttr := assertion.FindElement(".//Attribute[@Name='urn:oasis:names:tc:xacml:2.0:subject:role']//AttributeValue/Role")
+	if roleAttr == nil {
+		return nil, fmt.Errorf("role element not found in assertion")
+	}
+	roleAttr.CreateAttr("code", "182777000")
+
+	//Remove the workflow id if not provided
+	if launchContext.WorkflowId == "" {
```

Comment:
just don't add it in the first place

Reply:
We don't want to allow for a HCP token without a workflow though. For the application token, this is "allowed", for example when POST'ing a Flag

Reply:
Yeah definitely! I've refactored it, only optional for the application token now.